### PR TITLE
Add function for time sensitive aggregation

### LINKED ISSUE #1259: Add function to make time sensitive aggregations
- Issue type: Feature request

### Short Description
Parent issue: #1086.

As of now, the `ElasticsearchClient` can make aggregations for complete index only. This addition would make it possible to perform aggregations based on time of creation of messages.

### PR DESCRIPTION
**Pull request by singhpratyush**
### Short description

Fixes #1259.

I have:
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only strictly only one commit per issue.

### For the reviewers
I have:
- [ ] Reviewed this pull request by an authorized contributor.
- [ ] The reviewer is assigned to the pull request.

### CODE REVIEW

**Comment by singhpratyush** [comment]
@Achint08 @djmgit @hemantjadon @kavithaenair @SarishiNoHara @SKrPl @vibhcool: Please review.

File: src/org/loklak/data/ElasticsearchClient.java [hunk 1]
```diff
@@ -988,4 +990,65 @@ public TermsBuilder getClassifierAggregation(String classifierName) {
                 AggregationBuilders.sum("sum_probability").field(probabilityField)
             );
     }
+
+
```
Comment (vibhcool) [inline src/org/loklak/data/ElasticsearchClient.java]: extra space here

File: src/org/loklak/data/ElasticsearchClient.java [hunk 2]
```diff
@@ -988,4 +990,65 @@ public TermsBuilder getClassifierAggregation(String classifierName) {
...
+    public HashMap<String, HashMap<String, Double>> classifierScore(String index, String classifierName, ArrayList<String> classes, String startDate, String endDate) {
```
Comment (vibhcool) [inline src/org/loklak/data/ElasticsearchClient.java]: I have a doubt, does this method aggregate data acc. `to date` and acc. `to class` from query and then obtain some scores. What does class means here `sweat_smile`

Comment (singhpratyush) [inline src/org/loklak/data/ElasticsearchClient.java]: The top level data aggregation restricts the time to the desired range. The sub-aggregations added to it help perform the summation, averaging and document count jobs.

Comment (singhpratyush) [inline src/org/loklak/data/ElasticsearchClient.java]: The `classes` is a list of class that user wants information about.

File: src/org/loklak/data/ElasticsearchClient.java [hunk 3]
```diff
@@ -988,4 +990,64 @@ public TermsBuilder getClassifierAggregation(String classifierName) {
...
+            .setSize(0)
+            .addAggregation(getClassifierAggregation(classifierName, startDate, endDate));
+
+        SearchResponse response = request.execute().actionGet();
+        Range aggrs = response.getAggregations().get("by_date");
+
+        HashMap<String, HashMap<String, Double>> aggregatedMap = new HashMap<>();
+
+        for (Range.Bucket bkt : aggrs.getBuckets()) {
+            Terms termAggrs = bkt.getAggregations().get("by_class");
```
Comment (vibhcool) [inline src/org/loklak/data/ElasticsearchClient.java]: repeated declaration!!

Comment (singhpratyush) [inline src/org/loklak/data/ElasticsearchClient.java]: This is the best scope for this variable.

File: src/org/loklak/data/ElasticsearchClient.java [hunk 4]
```diff
@@ -988,4 +990,64 @@ public TermsBuilder getClassifierAggregation(String classifierName) {
...
+                map.put("avg", avg.getValue());
+                aggregatedMap.put(key,  map);
+            }
+        }
+        return aggregatedMap;
+    }
+
+    public AggregationBuilder getClassifierAggregation(String classifierName, String fromDate, String toDate) {
+        String probabilityField = classifierName + "_probability";
+        DateRangeBuilder db = AggregationBuilders.dateRange("by_date").field("created_at");
```
Comment (vibhcool) [inline src/org/loklak/data/ElasticsearchClient.java]: if the variables give some hint what it does would be better like filterData, etc (but not too big variables). Please see at other places too.

Comment (vibhcool) [inline src/org/loklak/data/ElasticsearchClient.java]: this one is suggestion, if you want 😅 :)

File: src/org/loklak/data/ElasticsearchClient.java (hunk 3 above)
Comment (vibhcool) [inline src/org/loklak/data/ElasticsearchClient.java]: ok, this answer says this is best scope :)
https://stackoverflow.com/questions/407255/difference-between-declaring-variables-before-or-in-loop

Comment (hemantjadon) [inline src/org/loklak/data/ElasticsearchClient.java]: I guess this is correct, and it really doesn't matter performance wise, but Namespace pollution is bad, so I guess this is perfect.

**Pull request merged**